artichoke-backend/src/extn/core/time/subsec.rs

```diff
@@ -0,0 +1,557 @@
+use crate::convert::implicitly_convert_to_int;
+use crate::extn::core::symbol::Symbol;
+///! Parser of Ruby Time subsecond parameters to help generate `Time`.
+///!
+///! This module implements the logic to parse two optional parameters in the
+///! `Time.at` function call. These parameters (if specified) provide the number
+///! of subsecond parts to add, and a scale of those subsecond parts (millis, micros,
+///! and nanos).
+use crate::extn::prelude::*;
+
+const NANOS_IN_SECOND: i64 = 1_000_000_000;
+
+const MILLIS_IN_NANO: i64 = 1_000_000;
+const MICROS_IN_NANO: i64 = 1_000;
+const NANOS_IN_NANO: i64 = 1;
+
+#[allow(clippy::cast_precision_loss)]
+const MIN_FLOAT_SECONDS: f64 = i64::MIN as f64;
+#[allow(clippy::cast_precision_loss)]
+const MAX_FLOAT_SECONDS: f64 = i64::MAX as f64;
+const MIN_FLOAT_NANOS: f64 = 0.0;
+#[allow(clippy::cast_precision_loss)]
+const MAX_FLOAT_NANOS: f64 = NANOS_IN_SECOND as f64;
+
+enum SubsecMultiplier {
```

Comment:
This enum is luckily not exposed anywhere (yet). Is it worthwhile throwing docs on it?

Reply:
I don't think so. Internal types haven't been getting docs historically.